serde/src/de/value.rs

```diff
@@ -22,31 +23,39 @@ use bytes;
 #[derive(Clone, Debug, PartialEq)]
 pub enum Error {
     /// The value had some syntatic error.
-    SyntaxError,
+    Syntax(String),
+
+    /// The value had an incorrect type.
+    Type(de::Type),
+
+    /// The value had an invalid length.
+    Length(usize),
```

Comment:
where would this be raised? When a 3 element tuple is expected but a 4 element sequence is given?